packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,226 @@
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+
+        uint256 startTime = challengeStartTime[msg.sender];
+        if (startTime == 0) {
+            revert LivenessModule2_ChallengeDoesNotExist();
+        }
+
+        // Check if response period has expired
+        if (block.timestamp >= startTime + config.livenessResponsePeriod) {
+            revert LivenessModule2_ResponsePeriodEnded();
+        }
+
+        delete challengeStartTime[msg.sender];
+        emit ChallengeCancelled(msg.sender);
+    }
+
+    /// @notice With a successful challenge, removes all current owners from an enabled safe, appoints fallback as its
+    /// sole owner, and sets its quorum to 1
+    /// @dev MUST be executable by anyone
+    /// @dev MUST revert if the given safe hasn't enabled the module
+    /// @dev MUST revert if there isn't a successful challenge for the given safe
+    /// @dev MUST enable the module to start a new challenge
+    /// @dev MUST emit the ChallengeExecuted event
+    /// @param _safe The Safe to transfer ownership of
+    function changeOwnershipToFallback(address _safe) external {
+        ModuleConfig storage config = safeConfigs[_safe];
+
+        if (config.fallbackOwner == address(0)) {
+            revert LivenessModule2_ModuleNotConfigured();
+        }
+
+        // Check that the module is still enabled on the target Safe
+        Safe safe = Safe(payable(_safe));
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        uint256 startTime = challengeStartTime[_safe];
+        if (startTime == 0) {
+            revert LivenessModule2_ChallengeDoesNotExist();
+        }
+
+        // Check if response period has expired
+        if (block.timestamp < startTime + config.livenessResponsePeriod) {
+            revert LivenessModule2_ResponsePeriodActive();
+        }
+
+        Safe targetSafe = Safe(payable(_safe));
+
+        // Get current owners
+        address[] memory owners = targetSafe.getOwners();
+
+        // Remove all owners after the first one
+        while (owners.length > 1) {
+            // removeOwner automatically updates the threshold, so we don't need to do it manually
+            targetSafe.execTransactionFromModule({
+                to: _safe,
+                value: 0,
+                operation: Enum.Operation.Call,
+                data: abi.encodeCall(OwnerManager.removeOwner, (SENTINEL_OWNER, owners[0], 1))
+            });
+            // Get updated owners list after removal
+            owners = targetSafe.getOwners();
+        }
+
+        // Now swap the remaining single owner with the fallback owner
+        targetSafe.execTransactionFromModule({
+            to: _safe,
+            value: 0,
+            operation: Enum.Operation.Call,
+            data: abi.encodeCall(OwnerManager.swapOwner, (SENTINEL_OWNER, owners[0], config.fallbackOwner))
+        });
```

Comment:
Can we add a sanity check here (similar to [this one](https://github.com/ethereum-optimism/optimism/blob/f67131314d235c30615eee3db113f6ecb9218f1b/packages/contracts-bedrock/src/safe/LivenessModule.sol#L244)) to ensure ownership is as expected?

Reply:
should ensure the fallback owner is the only owner.

Reply:
Added in a8a54cd0de6527e08029829d1a614e2e8c26209a